toolkit/offchain/src/register.rs

```diff
@@ -208,6 +295,30 @@ fn register_tx(
 	tx_builder.balance_update_and_build(ctx)
 }
 
+fn deregister_tx(
+	validator: &PlutusScript,
+	own_registration_utxos: &[OgmiosUtxo],
+	ctx: &TransactionContext,
+	validator_redeemer_ex_units: ExUnits,
+) -> Result<Transaction, JsError> {
+	let config = crate::csl::get_builder_config(ctx)?;
+	let mut tx_builder = TransactionBuilder::new(&config);
+
+	{
+		let mut inputs = TxInputsBuilder::new();
+		for own_registration_utxo in own_registration_utxos {
+			inputs.add_script_utxo_input(
+				own_registration_utxo,
+				validator,
+				validator_redeemer_ex_units.clone(),
```

Comment:
Is this right? If there were multiple utxos, we'd probably get multiple ex_units, no?

Reply:
It looks like the `validator_redeemer_ex_units` should be a vec, and it should get zipped with the utxos vector

Reply:
This is a good remark, but why zipped?  I'm not sure we have guarantees about order of adding inputs here and the order of costs returned by Ogmios.
If anything I would give each redeemer a different index and then match with the response from Ogmios.
I think we have exceeded the limit of expecting that we control order well enough..